test/common/config/registry_test.cc

```diff
@@ -182,6 +193,50 @@ TEST(RegistryTest, TestDoubleRegistrationByName) {
                             "Double registration for name: 'testing.published.test'");
 }
 
+class TestVersionedWithDeprecatedNamesFactoryAndAdditionalCategory : public PublishedFactory {
+public:
+  std::string category() const override { return "testing.published.additional.category"; }
+  std::string name() const override {
+    return "testing.published.versioned.instead_name_and_category";
+  }
+};
+
+TEST(RegistryTest, DEPRECATED_FEATURE_TEST(VersionedWithDeprecatedNamesFactoryAndNewCategory)) {
+  TestVersionedWithDeprecatedNamesFactoryAndAdditionalCategory test;
+
+  // Check the category is not registered
+  EXPECT_FALSE(Registry::FactoryCategoryRegistry::isRegistered(test.category()));
```

Comment:
This should be an ASSERT since otherwise registration will fail.

Reply:
sg